---
created: '2021-03-09T19:19:09.826292Z'
creator: sentry
source: tests/sentry/grouping/test_variants.py
---
app-depth-1:
  hash: "eb87c1031dba55b67df86fb9fff59dc6"
  tree_label: "log_demo::main"
  component:
    app-depth-1*
      exception*
        stacktrace*
          frame* (marked in-app by stack trace rule (family:native function:log_demo::* +app))
            function*
              "log_demo::main"
        type*
          "log_demo"
        value (ignored because stacktrace takes precedence)
          "Holy shit everything is on fire!"
--------------------------------------------------------------------------
app-depth-2:
  hash: "00cf41c4907b1c15c20189007fd0cffd"
  tree_label: "std::rt::lang_start::{{closure}}"
  component:
    app-depth-2*
      exception*
        stacktrace*
          frame* (marked out of app by stack trace rule (family:native function:std::* -app))
            function*
              "std::rt::lang_start::{{closure}}"
          frame* (marked in-app by stack trace rule (family:native function:log_demo::* +app))
            function*
              "log_demo::main"
          frame*
            function*
              "log::__private_api_log"
        type*
          "log_demo"
        value (ignored because stacktrace takes precedence)
          "Holy shit everything is on fire!"
--------------------------------------------------------------------------
app-depth-3:
  hash: "804fe59ad967ff79f51148c2fc1e1c2c"
  tree_label: "std::panicking::try::do_call"
  component:
    app-depth-3*
      exception*
        stacktrace*
          frame* (marked out of app by stack trace rule (family:native function:std::* -app))
            function*
              "std::panicking::try::do_call"
          frame* (marked out of app by stack trace rule (family:native function:std::* -app))
            function*
              "std::rt::lang_start::{{closure}}"
          frame* (marked in-app by stack trace rule (family:native function:log_demo::* +app))
            function*
              "log_demo::main"
          frame*
            function*
              "log::__private_api_log"
        type*
          "log_demo"
        value (ignored because stacktrace takes precedence)
          "Holy shit everything is on fire!"
--------------------------------------------------------------------------
app-depth-4:
  hash: "7c63b39cc8c54948f95871205ac3aeb5"
  tree_label: "___rust_maybe_catch_panic"
  component:
    app-depth-4*
      exception*
        stacktrace*
          frame*
            function*
              "___rust_maybe_catch_panic"
          frame* (marked out of app by stack trace rule (family:native function:std::* -app))
            function*
              "std::panicking::try::do_call"
          frame* (marked out of app by stack trace rule (family:native function:std::* -app))
            function*
              "std::rt::lang_start::{{closure}}"
          frame* (marked in-app by stack trace rule (family:native function:log_demo::* +app))
            function*
              "log_demo::main"
          frame*
            function*
              "log::__private_api_log"
        type*
          "log_demo"
        value (ignored because stacktrace takes precedence)
          "Holy shit everything is on fire!"
--------------------------------------------------------------------------
app-depth-5:
  hash: "7ea99ce5ec82e43f7b68aa44d57c0ee7"
  tree_label: "std::rt::lang_start_internal"
  component:
    app-depth-5*
      exception*
        stacktrace*
          frame* (marked out of app by stack trace rule (family:native function:std::* -app))
            function*
              "std::rt::lang_start_internal"
          frame*
            function*
              "___rust_maybe_catch_panic"
          frame* (marked out of app by stack trace rule (family:native function:std::* -app))
            function*
              "std::panicking::try::do_call"
          frame* (marked out of app by stack trace rule (family:native function:std::* -app))
            function*
              "std::rt::lang_start::{{closure}}"
          frame* (marked in-app by stack trace rule (family:native function:log_demo::* +app))
            function*
              "log_demo::main"
          frame*
            function*
              "log::__private_api_log"
        type*
          "log_demo"
        value (ignored because stacktrace takes precedence)
          "Holy shit everything is on fire!"
--------------------------------------------------------------------------
app-depth-max:
  hash: "e0b4eea234ff891472cb927c00153bbe"
  tree_label: "_main"
  component:
    app-depth-max*
      exception*
        stacktrace*
          frame*
            function*
              "_main"
          frame* (marked out of app by stack trace rule (family:native function:std::* -app))
            function*
              "std::rt::lang_start_internal"
          frame*
            function*
              "___rust_maybe_catch_panic"
          frame* (marked out of app by stack trace rule (family:native function:std::* -app))
            function*
              "std::panicking::try::do_call"
          frame* (marked out of app by stack trace rule (family:native function:std::* -app))
            function*
              "std::rt::lang_start::{{closure}}"
          frame* (marked in-app by stack trace rule (family:native function:log_demo::* +app))
            function*
              "log_demo::main"
          frame*
            function*
              "log::__private_api_log"
        type*
          "log_demo"
        value (ignored because stacktrace takes precedence)
          "Holy shit everything is on fire!"
--------------------------------------------------------------------------
system:
  hash: "e0b4eea234ff891472cb927c00153bbe"
  tree_label: "_main"
  component:
    system*
      exception*
        stacktrace*
          frame*
            function*
              "_main"
          frame* (marked out of app by stack trace rule (family:native function:std::* -app))
            function*
              "std::rt::lang_start_internal"
          frame*
            function*
              "___rust_maybe_catch_panic"
          frame* (marked out of app by stack trace rule (family:native function:std::* -app))
            function*
              "std::panicking::try::do_call"
          frame* (marked out of app by stack trace rule (family:native function:std::* -app))
            function*
              "std::rt::lang_start::{{closure}}"
          frame* (marked in-app by stack trace rule (family:native function:log_demo::* +app))
            function*
              "log_demo::main"
          frame*
            function*
              "log::__private_api_log"
        type*
          "log_demo"
        value (ignored because stacktrace takes precedence)
          "Holy shit everything is on fire!"
